### PR TITLE
kubectx: unpin go 1.17

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/kubectx/default.nix
Comment (superherointj) [inline pkgs/development/tools/kubectx/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
```
fetchpatch is not used?

**Review by superherointj — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/kubectx/default.nix
Comment (zowoq) [inline pkgs/development/tools/kubectx/default.nix:15]: Please submit this upstream.
```suggestion
    ./bump-golang-x-sys.patch
```

**Comment by stehessel** [comment]
Can you explain why this is superior to the version pin? The `0.9.4` release only guarantees Go compatibility up to `1.14`, see https://github.com/ahmetb/kubectx/blob/v0.9.4/go.mod#L3. So really we should pin the build to go `1.14` in a perfect world. Note that even master is only guaranteed to be compatible with `1.17` (https://github.com/ahmetb/kubectx/blob/master/go.mod#L3).

If we go the patch route, it should be upstreamed and then we can update the derivation to the new release.

File: pkgs/development/tools/kubectx/default.nix
Comment (SuperSandro2000) [reply pkgs/development/tools/kubectx/default.nix:15]: I did that but it won't apply because the go.mod changed otherwise since then in master https://github.com/ahmetb/kubectx/pull/360

Comment (zowoq) [reply pkgs/development/tools/kubectx/default.nix:15]: Maybe just switch to using unstable/master?

Comment (SuperSandro2000) [reply pkgs/development/tools/kubectx/default.nix:15]: I don't know, I leave that to the maintainer

**Comment by SuperSandro2000** [comment]
> The `0.9.4` release only guarantees Go compatibility up to `1.14`

That is not what this field is about. Please read up what it does on https://go.dev/doc/modules/gomod-ref#go . It is actually quite the opposite and means the module will (probably) only work on that go version or a newer one.

> So really we should pin the build to go `1.14` in a perfect world.

No, we would miss a massive amount of improvements, security fixes and platform support.

> If we go the patch route, it should be upstreamed
https://github.com/ahmetb/kubectx/pull/360

> and then we can update the derivation to the new release.

Upstream will probably not immediately release a new version and that is fine. The change is quite low level and super safe to merge.

**Comment by stehessel** [comment]
> That is not what this field is about. Please read up what it does on https://go.dev/doc/modules/gomod-ref#go . It is actually quite the opposite and means the module will (probably) only work on that go version or a newer one.

Thanks, I didn't know that.

**Pull request merged**
